torchgeo/datasets/mmflood.py

```diff
@@ -0,0 +1,422 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+"""MMFlood dataset."""
+
+import os
+import pathlib
+from collections.abc import Callable
+from glob import glob
+from typing import ClassVar, Literal, cast
+
+import matplotlib.pyplot as plt
+import numpy as np
+import pandas as pd
+import torch
+from matplotlib.figure import Figure
+from rasterio.crs import CRS
+from torch import Tensor
+
+from .errors import DatasetNotFoundError
+from .geo import RasterDataset
+from .utils import BoundingBox, Path, download_url, extract_archive
+
+
+class MMFlood(RasterDataset):
+    """MMFlood dataset.
+
+    `MMFlood <https://huggingface.co/datasets/links-ads/mmflood>`__ dataset is a multimodal flood delineation dataset.
+
+    Dataset features:
```

Comment:
Do you know how many different tiles there are for this dataset?  I think this would be useful as an indication of the dataset size. Is it one tile per flood event?

Reply:
Thanks for all your comments! I'll fix all the other parts as soon as possible.

About this, I don't know the exact number of unique satellite tiles that were downloaded to create this dataset... The authors of this dataset may have merged different tiles to obtain the 1748 smaller tif files (mentioned as acquisitions in the docstring) that are present in the dataset. Maybe @edornd has more details about this

Reply:
Hi @nilsleh, unfortunately we don't have the records of the individual tiles anymore since we built a mosaic on the spot when downloading the individual AoIs from Copernicus EMS. I confirm that there are 1748 individual AoI with varying sizes. Roughly, each AoI corresponds to a single S1 tile, since they're smaller, altough I remember that some overlapped. Each tile has an acquisition date tag though, so it is theoretically possible to backtrack the unique S1 tile ID.
When we trained for our benchmarks, we tiled these images into overlapping 512x512 chips obtaining 6182 chips for training, and 560 chips for validation (the test we tiled online at inference time), although these numbers are arbitrary and depend on implementation.

Reply:
Hi @edornd and @lccol, thanks for the reply. That's fine! I think the 1748 number is information enough for documentation purposes. So I think I would just add a bullet point to the class docstring along the lines of "1748 tiles of varying pixel dimensions".